### PR TITLE
Features/39

### PR DESCRIPTION
**Pull request by frantsao**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions
* Remember to set **idealista:develop** as base branch;

### Description of the Change
* Improved adding extra repositories

### Benefits
The role can be used to add other repositories than the distribution ones.

### Possible Drawbacks
N/A

### Applicable Issues
#39

### CODE REVIEW

File: molecule/corretto/group_vars/openjdk/main.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+
+java_open_jdk_version_major: 11
+# Supported versions: 8 (1.8.0 in RHEL), 11
+
+# For RHEL family
+java_required_repositories_openjdk:
+  - { name: "AmazonCoretto", baseurl: "https://yum.corretto.aws/$basearch" }
```
Comment (dortegau) [inline molecule/corretto/group_vars/openjdk/main.yml:8]: There is a typo: "AmazonCo**re**tto" -> "Amazon Corretto"

File: molecule/default/verify.yml [hunk 2]
```diff
@@ -31,12 +30,20 @@
...
     # Allowing to 'override' variables that are defined using include_vars
+
     - name: Java | Setting OS specific variables
       set_fact:
         java_open_jdk_version: "{{ java_open_jdk_version if java_open_jdk_version is defined and java_open_jdk_version is not sameas None }}"
         java_open_jdk_version_major: "{{ java_open_jdk_version_major if java_open_jdk_version_major is defined and java_open_jdk_version_major is not sameas None and java_open_jdk_version_major else __java_open_jdk_version_major }}"  # noqa 204
-        java_required_repositories_openjdk: "{{ java_required_repositories_openjdk if java_required_repositories_openjdk is defined and java_required_repositories_openjdk is not sameas None and java_required_repositories_openjdk|length else __java_required_repositories_openjdk }}"  # noqa 204
-        java_required_libs_openjdk: "{{ java_required_libs_openjdk if java_required_libs_openjdk is defined and java_required_libs_openjdk is not sameas None and java_required_libs_openjdk|length else __java_required_libs_openjdk }}"  # noqa 204
+
+    - name: Java | Setting OS specific variables (II)
```
Comment (dortegau) [inline molecule/default/verify.yml:39]: There aren't semantic/functional differences between the tasks describe in L:34, L:41 and L:44. I would collapse them in a single one

Comment (frantsao) [reply molecule/default/verify.yml:39]: Unfortunately, here the order is important. If I collapse all the set facts in the same task, all the vars are evaluated "at the same time", and crashes with "variable not defined" errors.
I'm investigating on alternative approachs, but it is the best solution by now.

File: defaults/main.yml [hunk 3]
```diff
@@ -1,7 +1,3 @@
 ---
-
-# OpenJDK implementation
-java_open_jdk_package: openjdk-{{ java_open_jdk_version_major }}-jdk-headless
-java_open_jdk_home_dir: java-{{ java_open_jdk_version_major }}-openjdk-amd64
+apt_extra_packages: []
```
Comment (dortegau) [inline defaults/main.yml:2]: I would recommend to add a prefix to this var in order to avoid possible collisions with other roles

**Pull request merged**